extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/vertx/VertxTracingAdapter.java

```diff
@@ -0,0 +1,293 @@
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanBuilder;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.context.propagation.TextMapGetter;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import io.opentelemetry.context.propagation.TextMapSetter;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import io.vertx.core.Context;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.core.http.HttpVersion;
+import io.vertx.core.impl.ContextInternal;
+import io.vertx.core.spi.VertxTracerFactory;
+import io.vertx.core.spi.tracing.SpanKind;
+import io.vertx.core.spi.tracing.TagExtractor;
+import io.vertx.core.spi.tracing.VertxTracer;
+import io.vertx.core.tracing.TracingOptions;
+import io.vertx.core.tracing.TracingPolicy;
+
+public class VertxTracingAdapter extends TracingOptions implements VertxTracer<Span, Span>, VertxTracerFactory {
+    private static final String SCOPE_KEY = VertxTracingAdapter.class.getName() + ".scope";
+    private static final String SPAN_KEY = VertxTracingAdapter.class.getName() + ".activeSpan";
+    private static TextMapPropagator TEXT_MAP_PROPAGATOR;
+
+    private Tracer tracer;
+
+    public void init() {
+        tracer = CDI.current().select(Tracer.class).get();
+
+        TEXT_MAP_PROPAGATOR = GlobalOpenTelemetry.getPropagators().getTextMapPropagator();
+    }
+
+    // TracingOptions overrides
+    @Override
+    public VertxTracerFactory getFactory() {
+        return this;
+    }
+
+    // VertxTracerFactory overrides
+    @Override
+    public VertxTracer<Span, Span> tracer(final TracingOptions options) {
+        return this;
+    }
+
+    // VertxTracer overrides
+    @Override
+    public <R> Span receiveRequest(
+            final Context context,
+            final SpanKind kind,
+            final TracingPolicy policy,
+            final R request,
+            final String operation,
+            final Iterable<Map.Entry<String, String>> headers,
+            final TagExtractor<R> tagExtractor) {
+
+        ((ContextInternal) context).dispatch(() -> {
+            io.opentelemetry.context.Context currentContext = current();
+
+            // Retrieve any incoming Span
+            io.opentelemetry.context.Context propagatedContext = TEXT_MAP_PROPAGATOR.extract(currentContext, headers, GETTER);
+
+            SpanBuilder builder;
+
+            // Create new span
+            builder = tracer.spanBuilder(operationName(request, operation))
+                    .setParent(propagatedContext)
+                    .setSpanKind(SpanKind.RPC.equals(kind) ? io.opentelemetry.api.trace.SpanKind.SERVER
+                            : io.opentelemetry.api.trace.SpanKind.CONSUMER);
+
+            //TODO - Figure out how to handle span name in a better way.
```

Comment:
Isn't the span name given by `operationName`? Is there any convention on how a span should be named?

Reply:
`operationName` is driven by Vert.x, which uses the OpenTracing naming conventions, and we need the OpenTelemetry semantic conventions.

OpenTelemetry conventions are here: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md

We still need to fix the cardinality of the name as at present we're not removing parameter values, but that's up next